lib/TopStoriesFeed.jsm

```diff
@@ -57,17 +55,14 @@ this.TopStoriesFeed = class TopStoriesFeed {
       // Cache is used for new page loads, which shouldn't have changed data.
       // If we have changed data, cache should be cleared,
       // and last updated should be 0, and we can fetch.
-      // Think there is a bug here.
       await this.loadCachedData();
       if (this.storiesLastUpdated === 0) {
-        shouldBroadcast = true;
         await this.fetchStories();
       }
       if (this.topicsLastUpdated === 0) {
-        shouldBroadcast = true;
         await this.fetchTopics();
       }
-      this.doContentUpdate(shouldBroadcast);
+      this.doContentUpdate(true);
```

Comment:
Essentially, this is how the old code worked, and I misinterpreted it when I was moving some stuff around.